### PR TITLE
tiny update

### PR DESCRIPTION
**Pull request by siddontang**
If we meet error, we should close the connection, not ignore the error and only log it. 

@ngaut @BusyJay @disksing

### CODE REVIEW

File: src/server/server.rs [hunk 1]
```diff
@@ -155,6 +155,9 @@ impl<T: RaftStoreRouter, S: StoreAddrResolver> Server<T, S> {
                 debug!("remove connection token {:?}", token);
                 // if connected to remote store, remove this too.
                 if let Some(store_id) = conn.store_id {
+                    warn!("remove store connection for store {} with token {:?}",
```
Comment (BusyJay) [inline src/server/server.rs]: I think info is enough.

Comment (siddontang) [inline src/server/server.rs]: If the store connection created, we won't close it arbitrarily, so here warn is ok.

**Comment by ngaut** [comment]
LGTM

**Comment by disksing** [comment]
LGTM.

**Comment by tiancaiamao** [comment]
can this change fix server random restart test case? @siddontang

**Comment by siddontang** [comment]
No

**Pull request merged**